### PR TITLE
Concrete data stores core constructors

### LINKED ISSUE #250: Ctors in concrete
```aya
open data Crash
| ready
| go Crash {
  | ready => ready
}
```

Aya is unhappy with this code:

```
  2 | | ready
  3 | | go Crash {
  4 |   | ready => ready
          ^---^

Error: Attempting to use a definition`Crash`which is not yet typechecked
```

The error message is kinda wrong btw

### PR DESCRIPTION
**Pull request by ice1000**
Fixes #250 

https://www.bilibili.com/video/BV1V34y1R7TZ

bors r+

### CODE REVIEW

File: base/src/test/resources/failure/patterns/issues2/issue250.aya.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+In file $FILE:3:4 ->
+
+  1 | open data Crash
+  2 | | go Crash {
+  3 |   | ready => ready
+          ^---^
+
+Error: Attempting to use a definition `ready` which is not yet typechecked
```
Comment (imkiva) [inline base/src/test/resources/failure/patterns/issues2/issue250.aya.txt:8]: We need to generate tyck order of ctors   🐘 🐐

**Review by imkiva — COMMENTED** [review]
clicked the link out of curiosity and I forgot to turn off my video in the classroom 🤣

File: base/src/test/resources/failure/patterns/issues2/issue250.aya.txt (hunk 1 above)
Comment (ice1000) [reply base/src/test/resources/failure/patterns/issues2/issue250.aya.txt:8]: I have a branch for that, it's called future

**Comment by ice1000** [comment]
> clicked the link out of curiosity and I forgot to turn off the volume in the classroom 🤣 

Lmfao

File: base/src/test/resources/failure/patterns/issues2/issue250.aya.txt (hunk 1 above)
Comment (imkiva) [reply base/src/test/resources/failure/patterns/issues2/issue250.aya.txt:8]: How to deal with mutual ctors? Do you have any idea?

Comment (ice1000) [reply base/src/test/resources/failure/patterns/issues2/issue250.aya.txt:8]: Basically, data signatures first, function signatures second, data body third, function bodies last

**Pull request merged**